### PR TITLE
feat: add thanos for multi cluster support

### PR DESCRIPTION
**Pull request by ArchiFleKs**

### CODE REVIEW

File: modules/aws/thanos.tf [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+  ])
+  namespace = local.thanos["create_ns"] ? kubernetes_namespace.thanos.*.metadata.0.name[0] : local.thanos["namespace"]
+
+  depends_on = [
+    helm_release.kube-prometheus-stack
+  ]
+}
+
+resource "kubernetes_secret" "thanos-ca" {
+  count = (local.thanos["enabled"] && local.thanos["generate_ca"]) || (local.thanos["enabled"] && local.thanos["trusted_ca_content"] != null) ? 1 : 0
```
Comment (rguichard) [inline modules/aws/thanos.tf:381]: maybe `local.thanos["enabled"] && (local.thanos["generate_ca"]) || (local.thanos["trusted_ca_content"] != null) ? 1 : 0` works too (and is smaller) ?

**Review by rguichard — COMMENTED** [review]
Maybe move cert-manager related commits into a different PR and keep this one about Prometheus/Thanos.

Except that, it looks good to me.

**Comment by ArchiFleKs** [comment]
@rguichard should be ok now

**Review by rguichard — APPROVED** [review]
lgtm

**Pull request merged**